### PR TITLE
replace redundant fmt.Sprintf() with strconv

### PR DESCRIPTION
**Pull request by thaJeztah**
I kept running into some of these, so thought I'd open a PR to replace them across the board. There's some minor ones remaining, but most of them should be good now.

See individual commits for details.

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: daemon/config/config.go [hunk 1]
```diff
@@ -317,15 +317,15 @@ func GetConflictFreeLabels(labels []string) ([]string, error) {
 		if len(stringSlice) > 1 {
 			// If there is a conflict we will return an error
 			if v, ok := labelMap[stringSlice[0]]; ok && v != stringSlice[1] {
-				return nil, fmt.Errorf("conflict labels for %s=%s and %s=%s", stringSlice[0], stringSlice[1], stringSlice[0], v)
+				return nil, errors.Errorf("conflict labels for %s=%s and %s=%s", stringSlice[0], stringSlice[1], stringSlice[0], v)
```
Comment (samuelkarp) [inline daemon/config/config.go:320]: Any particular reason for changing out the stdlib `fmt.Errorf` with a third-party `errors.Errorf`?

File: client/build_prune.go [hunk 2]
```diff
@@ -38,7 +38,7 @@ func (cli *Client) BuildCachePrune(ctx context.Context, opts types.BuildCachePru
 	}
 
 	if err := json.NewDecoder(serverResp.body).Decode(&report); err != nil {
-		return nil, fmt.Errorf("Error retrieving disk usage: %v", err)
+		return nil, errors.Wrap(err, "error retrieving disk usage")
```
Comment (samuelkarp) [inline client/build_prune.go:41]: Same here; I think this use of `fmt.Errorf` is appropriate.

File: daemon/config/config.go (hunk 1 above)
Comment (thaJeztah) [reply daemon/config/config.go:320]: This package was already using `pkg/errors` for all other errors, so aligning with other errors produced. 

`pkg/errors` also gives us a stack-trace (something Go's own error package unfortunately doesn't give); I realise it was archived (but mostly "feature complete"), but we were considering to reach out to the author to see if they're open to handing over maintainership.

File: client/build_prune.go (hunk 2 above)
Comment (thaJeztah) [reply client/build_prune.go:41]: mostly same as the other one 😅

**Review by samuelkarp — APPROVED** [review]

File: pkg/pidfile/pidfile.go [hunk 3]
```diff
@@ -36,10 +36,10 @@ func New(path string) (*PIDFile, error) {
 		return nil, err
 	}
 	// Note MkdirAll returns nil if a directory already exists
-	if err := system.MkdirAll(filepath.Dir(path), os.FileMode(0755)); err != nil {
+	if err := system.MkdirAll(filepath.Dir(path), os.FileMode(0o755)); err != nil {
```
Comment (thaJeztah) [inline pkg/pidfile/pidfile.go:39]: Looks like casting to `os.FileMode` is redundant here as well; let me update this commit to remove it.

Comment (Ahmedgkanria) [reply pkg/pidfile/pidfile.go]: []()

**Comment by thaJeztah** [comment]
Actually, let me remove the `pkg/pidfile` commit from this one; I have some other changes in that package, so let me group those.

**Comment by thaJeztah** [comment]
done; moved to https://github.com/moby/moby/pull/44266

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**